ocs_ci/deployment/aws.py

```diff
@@ -436,6 +441,433 @@ def destroy_cluster(self, log_level="DEBUG"):
             verify_stack_deleted(stack_name)
 
 
+class AWSUPIRHELWORKERS(AWSUPI):
+    """
+    A class to handle AWS UPI with RHEL worker nodes
+    """
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(AWSUPIRHELWORKERS, self).__init__()
+        self.worker_vpc = None
+        self.worker_iam_role = None
+        self.worker_subnet = None
+        self.worker_security_group = None
+        self.worker_tag = None
+        self.cf = None
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        # A dict for holding instance Name to instance object mapping
+        self.rhel_worker_list = {}
+        self.rhel_worker_user = constants.EC2_USER
+        self.client = boto3.client(
+            'ec2', region_name=config.ENV_DATA['region']
+        )
+
+    def deploy_cluster(self, log_cli_level='DEBUG'):
+        """
+        Deploy cluster overridden function from parent
+
+        Args:
+            log_cli_level (str): log level
+        """
+        if config.ENV_DATA['skip_ocp_deployment']:
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+        else:
+            prev_ocs_flag = config.ENV_DATA['skip_ocs_deployment']
+            # deploy only OCP
+            config.ENV_DATA['skip_ocs_deployment'] = 'true'
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+            self.add_rhel_workers()
+            config.ENV_DATA['skip_ocs_deployment'] = prev_ocs_flag
+            config.ENV_DATA['skip_ocp_deployment'] = 'true'
+            super(AWSUPIRHELWORKERS, self).deploy_cluster(log_cli_level)
+            config.ENV_DATA['skip_ocp_deployment'] = 'false'
```

Comment:
I don't think we want to be messing with the configuration like this. It's very error prone and it is necessary because you are overriding `deploy_cluster`. I don't think we need to override this and instead should be overriding `deploy_ocp`. This way you can add the call to `self.add_rhel_workers()` to the end of that method and let `deploy_cluster` keep doing what it was doing.

I haven't tested this at all but I think it would be worth looking into:
```python
def deploy_ocp(self, log_cli_level='DEBUG'):
    super(AWSUPIRHELWORKERS, self).deploy_ocp(log_cli_level)
    self.add_rhel_workers()
```

Reply:
Done, moved everything inside AWSUPI class